MetaMask/metamask-extension
# Feat/18889/update buttonlink ts

## Explanation

* Fixes #18889 
* Fixes #20214 

- Update ButtonLink to TS
- Fix ButtonLink alignment issue ( this issue came from padding being added my user style agent so I added 0px to the ButtonBase as a reset) 
- Fix ButtonLink hover underline to not apply when loading
- Fix ButtonLink loading color to be the same color



## Screenshots/Screencaps



### Before
<img width="416" alt="Screenshot 2023-08-02 at 3 50 58 PM" src="https://github.com/MetaMask/metamask-extension/assets/26469696/cbe2b9d8-1ed1-4944-a1b1-a2939a07a2f7">
<img width="642" alt="Screenshot 2023-08-02 at 3 51 30 PM" src="https://github.com/MetaMask/metamask-extension/assets/26469696/0afabf0c-a1a2-48d4-93a3-a8d813e1ff40">



### After
<img width="337" alt="Screenshot 2023-08-02 at 3 51 41 PM" src="https://github.com/MetaMask/metamask-extension/assets/26469696/3c5a4d6a-804c-4b79-8e1c-c7adac23d08a">
<img width="574" alt="Screenshot 2023-08-02 at 3 51 51 PM" src="https://github.com/MetaMask/metamask-extension/assets/26469696/184dc694-84c9-47a2-a88d-da949831d16c">



## Manual Testing Steps
`yarn jest ui/components/component-library`

- Visit ButtonLink component on Storybook to confirm props work as expected.
- There should be no underline on hover when in loading state. 
- The color of the loading wheel should reflect the ButtonLink text color. 
- Make sure the ButtonLink text is aligned when ButtonLinkSize.Inherit is being used. 



## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
